pypdf/_page.py

```diff
@@ -1519,6 +1519,23 @@ def compressContentStreams(self) -> None:  # deprecated
         )
         self.compress_content_streams()
 
+    @property
+    def page_number(self) -> int:
+        """
+        Read-only property which return the page number with the pdf file.
+
+        Returns:
+            int : page number ; -1 if the page is not attached to a pdf
+        """
+        if self.indirect_reference is None:
+            return -1
```

Comment:
What do you think about returning `None` or raising an exception?

Reply:
to be clear it it coverts the case where a page has been "built" but not yet inserted.
I did not want to change the output type. Also it is consistent with line 1537